filters/auth/grantlogout.go

```diff
@@ -0,0 +1,198 @@
+//
+// grantLogout filter
+//
+// Used for revoking access/refresh tokens generated by the oauthGrant() filter.
+//
+
+package auth
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	GrantLogoutName           = "grantLogout"
+	revokeTokenKey            = "token"
+	revokeTokenTypeKey        = "token_type_hint"
+	refreshTokenType          = "refresh_token"
+	accessTokenType           = "access_token"
+	unsupportedTokenTypeError = "unsupported_token_type"
+)
+
+type grantLogoutSpec struct {
+	config OAuthConfig
+}
+
+type grantLogoutFilter struct {
+	config OAuthConfig
+}
+
+type revokeErrorResponse struct {
+	Error            string `json:"error"`
+	ErrorDescription string `json:"error_description"`
+}
+
+func (grantLogoutSpec) Name() string { return GrantLogoutName }
+
+func (s grantLogoutSpec) CreateFilter([]interface{}) (filters.Filter, error) {
+	return grantLogoutFilter(s), nil
+}
+
+func (f grantLogoutFilter) basicAuthHeader() (string, error) {
+	clientID := f.config.GetClientID()
+	if clientID == "" {
+		return "", errors.New("failed to create token revoke auth header: no client ID")
+	}
+
+	clientSecret := f.config.GetClientSecret()
+	if clientSecret == "" {
+		return "", errors.New("failed to create token revoke auth header: no client secret")
+	}
+
+	credentials := fmt.Sprint(clientID, ":", clientSecret)
```

Comment:
Why fmt.Sprint?
`credentials := clientID + ":" + clientSecret`

Reply:
✅  Done. This has been replaced with the `SetBasicAuth` method.